core/src/main/java/com/datastax/oss/driver/internal/core/context/NettyOptions.java

```diff
@@ -62,6 +66,45 @@
    */
   ByteBufAllocator allocator();
 
+  /**
+   * Sets the socket options on a newly created client bootstrap. This is invoked each time the
+   * driver opens a new channel.
+   *
+   * <p>Implementation note: this is very similar to {@link #afterBootstrapInitialized}; the only
+   * reason to have two separate methods is backward compatibility.
+   *
+   * @param bootstrap the bootstrap to configure.
+   * @param defaultProfile the default profile of the driver's configuration (where the values to
+   *     set will typically be read from).
+   * @since 4.1.0
+   */
+  default void setSocketOptions(Bootstrap bootstrap, DriverExecutionProfile defaultProfile) {
+    boolean tcpNoDelay = defaultProfile.getBoolean(DefaultDriverOption.SOCKET_TCP_NODELAY);
+    bootstrap.option(ChannelOption.TCP_NODELAY, tcpNoDelay);
```

Comment:
Do we want to use the same `isDefined()` check here for the Unix socket case, or is the idea that when using Unix sockets, you would provide a different implementation for `setSocketOptions` that does not try to set TCP_NODELAY?

Reply:
I commented too soon. We are doing it this way to keep the functionality backward compatible, correct?